components/reply.js

```diff
@@ -33,10 +34,32 @@ function FreebieDialog () {
   )
 }
 
-export default function Reply ({ item, onSuccess, replyOpen, children, placeholder }) {
+export default function Reply ({ item, onSuccess, replyOpen, children, placeholder, innerRef }) {
   const [reply, setReply] = useState(replyOpen)
   const me = useMe()
   const parentId = item.id
+  const replyInput = useRef(null)
+  const formInnerRef = useRef()
+  const quoteReply = useCallback(() => {
+    if (!reply) {
+      setReply(true)
+    }
+    let updatedValue
+    if (formInnerRef.current && formInnerRef.current.values && !formInnerRef.current.values.text) {
+      updatedValue = quote(item.text)
+    } else if (formInnerRef.current?.values?.text) {
+      // append quote reply text if the input already has content
+      updatedValue = `${replyInput.current.value}\n${quote(item.text)}`
+    }
+    if (updatedValue) {
+      replyInput.current.value = updatedValue
+      formInnerRef.current.setValues({ text: updatedValue })
+      window.localStorage.setItem(`reply-${parentId}-text`, updatedValue)
+    }
+  }, [reply, item])
+  if (innerRef) {
+    innerRef.current = { quoteReply }
```

Comment:
Apparently I just re-invented https://react.dev/reference/react/useImperativeHandle lol, I'll refactor to use that